pygfx/renderers/wgpu/_renderer.py

```diff
@@ -404,48 +433,41 @@ def render(
         # todo: also note that the fragment shader is (should be) optional
         #      (e.g. depth only passes like shadow mapping or z prepass)
 
-        # Get logical size (as two floats). This size is constant throughout
-        # all post-processing render passes.
-        target_size, logical_size = get_size_from_render_target(self._target)
-        if not all(x > 0 for x in logical_size):
+        # Get size for the render textures
+        logical_size = self.logical_size
+        physical_size = self.physical_size
+        if not all(i > 0 for i in logical_size):
             return
-
-        # Determine the physical size of the render texture
-        target_pixel_ratio = target_size[0] / logical_size[0]
-        if self._pixel_ratio:
-            pixel_ratio = self._pixel_ratio
-        else:
-            pixel_ratio = target_pixel_ratio
-            if pixel_ratio <= 1:
-                pixel_ratio = 2.0  # use 2 on non-hidpi displays
-
-        # Determine the physical size of the first and last render pass
-        framebuffer_size = tuple(max(1, int(pixel_ratio * x)) for x in logical_size)
+        pixel_ratio = physical_size[1] / logical_size[1]
 
         # Update the render targets
-        self._blender.ensure_target_size(device, framebuffer_size)
+        self._blender.ensure_target_size(device, physical_size)
+
+        # The viewpoer could be a Viewport object
+        if hasattr(viewport, "rect"):
+            viewport = viewport.rect
```

Comment:
propose to rename the kwarg `viewport` to `rect`, and always require a tuple to be passed (so no support for Viewport object to be passed in)